### PR TITLE
feat(ui5-avatar-group): Implement accessibility specification

### PR DESCRIPTION
**Pull request by dobrinyonkov**
Fixes: #2745

* new property `ariaHaspopup` added to `ui5-avatar` and `ui5-avatar-group`
* aria-haspopup attribute of the overflow button or the avatars container renders (or not) depending on the value of
the `ariaHaspopup` property and the `type` property
* interactive `ui5-avatar` now has role `button`

### CODE REVIEW

File: packages/main/src/Avatar.js [hunk 1]
```diff
@@ -196,6 +197,30 @@ const metadata = {
...
+		 * <ul>
+		 *
+		 * <li><code> None - the aria-haspopup attribute will not be rendered.</li>
+		 * <li><code> Menu - Indicates the popup is a menu.</li>
+		 * <li><code> Listbox - Indicates the popup is a listbox.</li>
+		 * <li><code> Tree - Indicates the popup is a tree.</li>
+		 * <li><code> Grid - Indicates the popup is a grid.</li>
+		 * <li><code> Dialog - Indicates the popup is a dialog.</li>
+		 *
+		 * <ul>
```
Comment (fifoosid) [inline packages/main/src/Avatar.js:213]: </ul>

File: packages/main/src/Avatar.js [hunk 2]
```diff
@@ -196,6 +197,30 @@ const metadata = {
 			type: String,
 		},
 
+		/**
+		 * Defines the aria-haspopup value of the <code>ui5-avatar</code> when <code>interactive</code> property is <code>true</code>.
+		 * <br><br>
+		 * Available options are:
+		 * <ul>
+		 *
+		 * <li><code> None - the aria-haspopup attribute will not be rendered.</li>
```
Comment (fifoosid) [inline packages/main/src/Avatar.js:206]: Add </code> to the end of every list item

File: packages/main/src/AvatarGroup.js [hunk 3]
```diff
@@ -89,6 +100,35 @@ const metadata = {
 			defaultValue: AvatarSize.S,
 		},
 
+		/**
+		 * Defines the aria-haspopup value of the <code>ui5-avatar-group</code> on:
+		 * <br><br>
+		 * <ul>
+		 * <li> the whole container when <code>type</code> property is <code>Group</code></li>
+		 * <li> the default "More" overflow button when <code>type</code> is <code>Individual</code></li>
+		 * <ul>
```
Comment (fifoosid) [inline packages/main/src/AvatarGroup.js:109]: </ul>

File: packages/main/src/AvatarGroup.js [hunk 4]
```diff
@@ -89,6 +100,35 @@ const metadata = {
...
+		 *
+		 * <li> None - the aria-haspopup attribute will not be rendered.</li>
+		 * <li> Menu - Indicates the popup is a menu.</li>
+		 * <li> Listbox - Indicates the popup is a listbox.</li>
+		 * <li> Tree - Indicates the popup is a tree.</li>
+		 * <li> Grid - Indicates the popup is a grid.</li>
+		 * <li> Dialog - Indicates the popup is a dialog.</li>
+		 *
+		 * <ul>
```
Comment (fifoosid) [inline packages/main/src/AvatarGroup.js:121]: </ul>

File: packages/main/src/AvatarGroup.js [hunk 5]
```diff
@@ -195,6 +235,30 @@ const metadata = {
  * <li>You want to display a gallery for simple images.</li>
  * <li>You want to use it for other visual content than avatars.</li>
  * </ul>
+ * 
+ * <h3>Keyboard Handling</h3>
+ * The <code>ui5-avatar-group</code> provides advanced keyboard handling.
+ * When focused the user can use the following keyboard
```
Comment (fifoosid) [inline packages/main/src/AvatarGroup.js:241]: When focused, the ...

**Review by fifoosid — CHANGES_REQUESTED** [review]
From code POV, looks good to me. I have left some small comments regarding the docs. You can fix the build as well, because there are some lint errors:
image

Once you fix the comments and get a review from @SAP/ui5-webcomponents-topic-p we can merge it.

**Review by fifoosid — COMMENTED** [review]
There is a merge conflict, which needs to be resolved

**Review by fifoosid — COMMENTED** [review]
Looks good to me, someone from @SAP/ui5-webcomponents-topic-p should approve and we can merge

**Review by ilhan007 — DISMISSED** [review]

**Review by ilhan007 — APPROVED** [review]

**Pull request merged**